frontend/src/metabase/querying/components/FilterModal/FilterModal.tsx

```diff
@@ -63,32 +71,34 @@ export function FilterModal({
     onClose();
   };
 
+  const handleSearch = (searchText: string) => {
+    setTab(isSearchActive(searchText) ? SEARCH_KEY : groupItems[0]?.key);
+    setSearchText(searchText);
+  };
+
   return (
     <Modal.Root opened size={getModalWidth(groupItems)} onClose={onClose}>
       <Modal.Overlay />
       <Modal.Content>
         <ModalHeader p="lg">
           <Modal.Title>{getModalTitle(groupItems)}</Modal.Title>
+          <FilterSearchInput onChange={handleSearch} />
```

Comment:
Having the input so close to an "x" button makes it look like the "x" button is for clearing the input.